File: POTD/Javascript/findAllDuplicatesInAnArray.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    const duplicates = [];
    for (let i = 0; i < nums.length; i++) {
        let index = Math.abs(nums[i]) - 1;
        if (nums[index] < 0) {
            duplicates.push(Math.abs(nums[i]));
        }
        nums[index] = -nums[index];
    }
    return duplicates;
};

File: POTD/Python/findAllDuplicatesInAnArray.py
class Solution:
  def findDuplicates(self, nums: List[int]) -> List[int]:
      duplicates = []
      for i in range(len(nums)):
          if nums[abs(nums[i])-1] < 0:
              duplicates.append(abs(nums[i]))
          nums[abs(nums[i])-1] = -1 * nums[abs(nums[i])-1]
      return duplicates